### PR TITLE
Change Data for chart & export and export data processing

### PR DESCRIPTION
**Pull request by MicheleK17**
- add translation for new data

Ticket #372

Change-Id: Ia620ba914de40617ac282370d7e2c39622d1b5d7

### CODE REVIEW

**Review by GerberI — CHANGES_REQUESTED** [review]
Commit: Refactor Table components
Good, refactored structure is now more clear and layouts are now correctly implemented.

File: src/app/Main/Body/QueryPage/Services/ExportServices/StatExportServices/generateStatisticRowsCsvString.service.ts [hunk 1]
```diff
@@ -13,11 +13,16 @@ export function generateStatisticRowsCsvString<
     const csvTable: string[] = [];
 
     dataArray.forEach((row: T) => {
-        const values: string[] = headers.map((header: K) => {
-            const rowValue = row[header];
-            return modifyTableDataStringService(`${rowValue}`);
+        const rowValues: string[] = [];
+        headers.forEach((header) => {
+            if (header === "name") {
+                rowValues.unshift(modifyTableDataStringService(row.name));
```
[the remark this answers is not in the document]
Comment (MicheleK17) [reply src/app/Main/Body/QueryPage/Services/ExportServices/StatExportServices/generateStatisticRowsCsvString.service.ts:19]: Yes, that seemed the safer way.

File: src/app/Main/Body/QueryPage/QueryPageContent/Results/TableResults/TableResults-Layout.component.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { AccordionComponent } from "../../../../../../Shared/Accordion.component";
+import { DataInterface } from "../../../../../../Shared/Context/DataContext";
+
+const dataStyle = css`
+    max-width: fit-content;
+    margin: auto;
+    box-sizing: inherit;
+`;
+
+export interface TableResultsProps {
```
[the remark this answers is not in the document]
Comment (MicheleK17) [reply src/app/Main/Body/QueryPage/QueryPageContent/Results/TableResults/TableResults-Layout.component.tsx]: Done.

File: src/app/Main/Body/QueryPage/QueryPageContent/Results/TableResults/Table/createTableRowsWithColSum.service.tsx [hunk 3]
```diff
@@ -25,12 +26,12 @@ const totalSumStyle = css`
     background-color: ${sumRowColBackgroundColor};
 `;
 
-export function TableContentRowWithColSumComponent(props: {
+export function createTableRowWithColSum(props: {
```
[the remark this answers is not in the document]
Comment (MicheleK17) [reply src/app/Main/Body/QueryPage/QueryPageContent/Results/TableResults/Table/createTableRowsWithColSum.service.tsx]: Done.

File: src/app/Main/Body/QueryPage/QueryPageContent/Results/TableResults/Table/Table-Layout.component.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    flex-direction: row;
+`;
+
+/**
+ * @desc Decides if row/colum is selected and return result table or explanation text
+ * @param columnAttributes - column attributes for the table header
+ * @returns {JSX.Element} - result table
+ */
+export function TableLayout(props: {
+    columnMainHeader: JSX.Element | undefined;
```
[the remark this answers is not in the document]
Comment (MicheleK17) [reply src/app/Main/Body/QueryPage/QueryPageContent/Results/TableResults/Table/Table-Layout.component.tsx]: Done.

File: src/app/Main/Body/QueryPage/QueryPageContent/Results/TableResults/TableResults-Container.component.tsx [hunk 5]
```diff
@@ -91,11 +80,24 @@ export function QueryPageContentTableResultsLayoutComponent(
...
+        );
+    }
+
     return (
-        <AccordionComponent
-            title={accordionHeader}
-            content={tableAccordionContent}
-            defaultExpanded
+        <TableResultsLayout
+            accordionHeader={accordionHeader}
```
[the remark this answers is not in the document]
Comment (MicheleK17) [reply src/app/Main/Body/QueryPage/QueryPageContent/Results/TableResults/TableResults-Container.component.tsx]: Done. Last suggestion.

File: src/locales/de/QueryPage.json [hunk 6]
```diff
@@ -155,6 +163,7 @@
             "Salm-Bekämpfung": "Salm-Bekämpfung"
         },
         "federalState": {
+            "keine Anzeige": "keine Anzeige",
```
[the remark this answers is not in the document]
Comment (MicheleK17) [reply src/locales/de/QueryPage.json]: Done.

**Review by GerberI — APPROVED** [review]
Commit: Little changes and move center style to accordion
Looks good, thx

**Pull request merged**